### PR TITLE
Changed some traffic_layout related codes to memory efficient string_view

### PR DESCRIPTION
**Pull request by chitianhao**
Some changes to std::string_view corresponding to previous closed PR #3580 

``lib/ts/ink_args``
``lib/ts/runroot.cc``
``cmd/traffic_layout/info.cc``

@SolidWallOfCode

### CODE REVIEW

File: lib/ts/ink_args.cc [hunk 1]
```diff
@@ -223,7 +223,7 @@ process_args_ex(const AppVersionInfo *appinfo, const ArgumentDescription *argume
       // Deal with long options ...
       for (i = 0; i < n_argument_descriptions; i++) {
         // handle the runroot arg
-        std::string cur_argv = *argv + 2;
+        std::string_view cur_argv = *argv + 2;
```
Comment (zwoop) [inline lib/ts/ink_args.cc]: Are we positive this has the same semantics? The std:;string copies (I assume) the string from argv, whereas string_view sill only have a pointer into the argv. Have you made sure this is safe in all places? The possible problem being if argv goes out of scope, but something still has  the string_view "pointer", you are now addressing garbage.

File: lib/ts/runroot.cc [hunk 2]
```diff
@@ -108,9 +108,9 @@ runroot_handler(const char **argv, bool json)
 
   int i = 0;
   while (argv[i]) {
-    std::string command = argv[i];
+    std::string_view command = argv[i];
```
Comment (zwoop) [inline lib/ts/runroot.cc]: Same here.

File: lib/ts/ink_args.cc (hunk 1 above)
Comment (chitianhao) [reply lib/ts/ink_args.cc]: There should be no issue with it since it is only used once immediately after this assigning in the following line.

Comment (dragon512) [reply lib/ts/ink_args.cc]: ya this is safe

**Comment by dragon512** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Cherry-picked to 8.0.x

**Pull request merged**